syncd/ServiceMethodTable.cpp

```diff
@@ -2,6 +2,7 @@
 
 #include "swss/logger.h"
 
+#include <array>
```

Comment:
Why this is added?

Reply:
GCC 12 complained that `std::array` was not completely defined, but is instantiated on line 74.

Snippet of error log:

```
ServiceMethodTable.cpp: In instantiation of 'constexpr auto declare_static(std::index_sequence<i ...>) [with B = syncd::ServiceMethodTable::SlotBase; D = syncd::ServiceMethodTable::Slot; long unsigned int ...i = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9}; std::index_sequence<i ...> = std::integer_sequence<long unsigned int, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9>]':
ServiceMethodTable.cpp:81:35:   required from 'constexpr auto declare_static() [with B = syncd::ServiceMethodTable::SlotBase; D = syncd::ServiceMethodTable::Slot; long unsigned int size = 10]'
ServiceMethodTable.cpp:86:79:   required from here
ServiceMethodTable.cpp:74:56: error: invalid use of incomplete type 'struct std::array<syncd::ServiceMethodTable::SlotBase*, 10>'
   74 |     return std::array<B*, sizeof...(i)>{{new D<i>()...}};
      |                                                        ^
In file included from /usr/include/c++/12/bits/stl_map.h:63,
                 from /usr/include/c++/12/map:61,
                 from /usr/include/swss/logger.h:6,
                 from ServiceMethodTable.h:7,
                 from ServiceMethodTable.cpp:1:
/usr/include/c++/12/tuple:1595:45: note: declaration of 'struct std::array<syncd::ServiceMethodTable::SlotBase*, 10>'
 1595 |   template<typename _Tp, size_t _Nm> struct array;
      |                                             ^~~~~
ServiceMethodTable.cpp: In instantiation of 'constexpr auto declare_static() [with B = syncd::ServiceMethodTable::SlotBase; D = syncd::ServiceMethodTable::Slot; long unsigned int size = 10]':
ServiceMethodTable.cpp:86:79:   required from here
ServiceMethodTable.cpp:82:48: error: no matching function for call to 'std::vector<syncd::ServiceMethodTable::SlotBase*>::vector(<brace-enclosed initializer list>)'
   82 |     return std::vector<B*>{arr.begin(), arr.end()};
      |                                         ~~~~~~~^~
```

Reply:
3hivh version cimpiler

Reply:
GCC 12.2, specifically (also using libstdc++ 12.2)

Reply:
If you want, I think I can modify the pipeline to build this on Bookworm as well, so that is covered.

Reply:
Your title suggest tat this is the change so support bookworm so what other changes would be needed ?

Reply:
Not sure at this point. At this point, only compilation (and the tests that get run as part of the build) have been done; runtime testing has not been done yet.

Reply:
Then please update title to reflect changes, like GCC latest support or update headers etc

Reply:
Updated